pkg/standalone-utils/test/ProtocolFeePercentagesProvider.test.ts

```diff
@@ -0,0 +1,470 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { BigNumber, BigNumberish, Contract } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import { deploy, deployedAt, getArtifact } from '@balancer-labs/v2-helpers/src/contract';
+import { ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+
+describe('ProtocolFeePercentagesProvider', function () {
+  let admin: SignerWithAddress, authorized: SignerWithAddress, other: SignerWithAddress;
+  let authorizer: Contract, vault: Contract, feesCollector: Contract;
+  let provider: Contract;
+
+  enum FeeType {
+    Swap = 0,
+    FlashLoan = 1,
+    Yield = 2,
+    AUM = 3,
+  }
+
+  const INVALID_FEE_TYPE = 1047;
+
+  // Note that these two values are not passed - they're hardcoded into the ProtocolFeesCollector
+  const MAX_SWAP_VALUE = fp(0.5);
+  const MAX_FLASH_LOAN_VALUE = fp(0.01);
+
+  const MAX_AUM_VALUE = fp(0.2);
+  const MAX_YIELD_VALUE = fp(0.8);
+
+  before(async () => {
+    [, admin, authorized, other] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy vault', async () => {
+    authorizer = await deploy('v2-vault/TimelockAuthorizer', { args: [admin.address, ZERO_ADDRESS, 0] });
+    vault = await deploy('v2-vault/Vault', { args: [authorizer.address, ZERO_ADDRESS, 0, 0] });
+    feesCollector = await deployedAt('v2-vault/ProtocolFeesCollector', await vault.getProtocolFeesCollector());
+  });
+
+  describe('construction', () => {
+    it('reverts if the maximum yield value is too high', async () => {
+      await expect(
+        deploy('ProtocolFeePercentagesProvider', {
+          args: [vault.address, fp(1).add(1), 0],
+        })
+      ).to.be.revertedWith('Invalid maximum fee percentage');
+    });
+
+    it('reverts if the maximum aum value is too high', async () => {
+      await expect(
+        deploy('ProtocolFeePercentagesProvider', {
+          args: [vault.address, 0, fp(1).add(1)],
+        })
+      ).to.be.revertedWith('Invalid maximum fee percentage');
+    });
+
+    it('emits ProtocolFeeTypeRegistered events for custom types', async () => {
+      // We deploy manually instead of using our helper to get the transaction receipt
```

Comment:
You can read this off from the returned contract using the `deployTransaction` field. We use this in the `deployments` package:

https://github.com/balancer-labs/balancer-v2-monorepo/blob/77efc90be71c5e44434287e67b1b58d037241f01/pkg/deployments/src/task.ts#L123

Reply:
Ahh nice, thank you!

Reply:
Note that we have two `deploy` helpers, which seem to work slightly differently (in one we return the object created by ethers, in the other we discard it and just keep the address to then create a new contract). I'll keep the tests as-is - we can merge the helpers later down the line.